apps/passport-server/src/database/queries/telegram/fetchTelegramEvent.ts

```diff
@@ -184,19 +185,74 @@ export async function fetchTelegramChatsWithMembershipStatus(
 
 export async function fetchTelegramTopic(
   client: Pool,
-  telegramChatId: number,
-  topicId: number
-): Promise<TelegramTopic | null> {
-  const result = await sqlQuery(
-    client,
-    ` 
-    SELECT * 
+  telegramChatId: number | string,
+  topicId?: number | string
+): Promise<TelegramTopicFetch | null> {
+  const query = `
+    SELECT telegram_chat_id AS "telegramChatID", * 
     FROM telegram_chat_topics
     WHERE 
         telegram_chat_id = $1 AND 
-        topic_id = $2
-  `,
-    [telegramChatId, topicId]
-  );
+        (topic_id = $2 OR ($2 IS NULL AND topic_id IS NULL))
+  `;
+  const result = await sqlQuery(client, query, [
+    telegramChatId,
+    topicId || null
+  ]);
   return result.rows[0] ?? null;
 }
+
+export async function fetchTelegramTopicsReceiving(
+  client: Pool
+): Promise<TelegramTopicWithFwdInfo[]> {
+  const result = await sqlQuery(
+    client,
+    ` 
+    SELECT telegram_chat_id AS "telegramChatID", * 
+    FROM telegram_chat_topics tct
+    JOIN telegram_forwarding tf ON tct.id = tf.receiver_chat_topic_id
+    WHERE tf.receiver_chat_topic_id IS NOT NULL;`,
+    []
+  );
+  return result.rows;
+}
+
+const linkedDestinationToTopic = (
+  row: any
+): TelegramTopicFetch & { forwardDestination: TelegramTopicFetch } => {
+  return {
+    ...row,
+    forwardDestination: {
+      id: row.forwardingID,
+      topic_id: row.forwardingTopicID,
+      topic_name: row.forwardingTopicName,
+      telegramChatID: row.forwardingChatID
+    }
+  };
+};
+
+export async function fetchTelegramTopicForwarding(
+  client: Pool,
+  telegramChatID: string | number,
+  topicId?: string | number
+): Promise<
+  (TelegramTopicFetch & { forwardDestination: TelegramTopicFetch })[]
+> {
+  const result = await sqlQuery(
+    client,
+    `
+    SELECT 
+      tct.telegram_chat_id AS "telegramChatID",
+      tct.*,
+      tf2.id AS "forwardingID",
+      tf2.topic_id AS "forwardingTopicID",
+      tf2.topic_name AS "forwardingTopicName",
+      tf2.telegram_chat_id AS "forwardingChatID"
+    FROM telegram_chat_topics tct
+    JOIN telegram_forwarding tf ON tct.id = tf.sender_chat_topic_id
+    LEFT JOIN telegram_chat_topics tf2 ON tf.receiver_chat_topic_id = tf2.id
+    WHERE tct.telegram_chat_id = $1 AND (tct.topic_id = $2 OR ($2 IS NULL AND tct.topic_id IS NULL)) AND tf.sender_chat_topic_id IS NOT NULL;`,
```

Comment:
this is a really big query that's not super clear how to parse. recommend adding a comment in the function signature explaining what's happening

Reply:
Good point, I made the query more obvious and added a comment